yarn-project/prover-node/src/prover-node-publisher.test.ts

```diff
@@ -0,0 +1,180 @@
+
+    // We try and prove all of the epoch - succeeds
+    { pendingBlockNumber: 65n, provenBlockNumber: 40n, fromBlock: 33, toBlock: 64, expectedPublish: true, message: '' },
+
+    // We try and partially prove the epoch after their proof - succeeds again
+    { pendingBlockNumber: 65n, provenBlockNumber: 40n, fromBlock: 41, toBlock: 45, expectedPublish: true, message: '' },
+
+    // We try and partially prove the epoch on top of their proof - succeeds again
+    { pendingBlockNumber: 65n, provenBlockNumber: 40n, fromBlock: 33, toBlock: 45, expectedPublish: true, message: '' },
+
+    // We try and partially prove the epoch and partially on top of their proof - succeeds again
+    { pendingBlockNumber: 65n, provenBlockNumber: 40n, fromBlock: 35, toBlock: 45, expectedPublish: true, message: '' },
+
+    // We try and partially prove the epoch but less than was already proven - fails as pointless
+    {
+      pendingBlockNumber: 65n,
+      provenBlockNumber: 40n,
+      fromBlock: 33,
+      toBlock: 39,
+      expectedPublish: false,
+      message: 'Cannot submit epoch proof for 33-39 as proven block is 40',
+    },
+
+    // We try and partially prove the epoch but the same as was already proven - should possibly fail but succeeds for now, quite an edge case
+    {
+      pendingBlockNumber: 65n,
+      provenBlockNumber: 40n,
+      fromBlock: 33,
+      toBlock: 40,
+      expectedPublish: true,
+    },
+  ];
+
+  test.each(testCases)(
+    'submits proof for epoch with pendingBlock: $pendingBlockNumber, provenBlock: $provenBlockNumber, fromBlock: $fromBlock, toBlock: $toBlock',
+    async ({ pendingBlockNumber, provenBlockNumber, fromBlock, toBlock, expectedPublish, message }) => {
+      // Create proofs for every block, to reference as previous archives
+      const blocks = Array.from({ length: 100 }, () => {
+        const publicInputs = RootRollupPublicInputs.random();
+        return {
+          publicInputs,
+          proof: Proof.empty(),
+        };
+      });
+
+      // Return the tips specified by the test
+      rollup.getTips.mockResolvedValue({
+        pendingBlockNumber,
+        provenBlockNumber,
+      });
+
+      // Return the requested block
+      rollup.getBlock.mockImplementation((blockNumber: bigint) =>
+        Promise.resolve({
+          blockHash: blocks[Number(blockNumber) - 1].publicInputs.endBlockHash.toString(),
+          archive: blocks[Number(blockNumber) - 1].publicInputs.endArchive.root.toString(),
+          slotNumber: 0n, // unused,
+        }),
+      );
+
+      const ourLastBlock = blocks[toBlock - 1];
+      const publicInputs = ourLastBlock.publicInputs;
+      publicInputs.previousBlockHash = blocks[fromBlock - 2]?.publicInputs.endBlockHash ?? Fr.ZERO;
+      publicInputs.previousArchive = blocks[fromBlock - 2]?.publicInputs.endArchive ?? Fr.ZERO;
+      const proof = ourLastBlock.proof;
```

Comment:
```suggestion
      const ourFirstBlock = blocks[fromBlock - 1];
      const publicInputs = ourFirstBlock.publicInputs;
      publicInputs.previousBlockHash = blocks[fromBlock - 2]?.publicInputs.endBlockHash ?? Fr.ZERO;
      publicInputs.previousArchive = blocks[fromBlock - 2]?.publicInputs.endArchive ?? Fr.ZERO;
      const proof = ourFirstBlock.proof;
```
Shouldn't we be tweaking the `previousArchive` of the first block we're proving?

Reply:
Cleaned the test up so should be clearer now